README.md

```diff
@@ -0,0 +1,180 @@
+  # For ease of use, `deploy-rs` provides a function to easy all this required activation script to any derivation
+  # Both the working directory and `$PROFILE` will point to `profilePath`
+  path = deploy-rs.lib.x86_64-linux.setActivate pkgs.hello "./bin/hello";
+
+  # An optional path to where your profile should be installed to, this is useful if you want to use a common profile name across multiple users, but would have conflicts in your node's profile list.
+  # This will default to `"/nix/var/nix/profiles/$PROFILE_NAME` if `user` is root (see: generic options), and `/nix/var/nix/profiles/per-user/$USER/$PROFILE_NAME` if it is not.
+  profilePath = "/nix/var/nix/profiles/per-user/someuser/someprofile";
+
+  # ...generic options... (see lower section)
+}
+```
+
+### Node
+
+This defines a single node/server, and the profiles you intend it to run.
+
+```nix
+{
+  # The hostname of your server, don't worry, this can be overridden at runtime if needed
+  hostname = "my.server.gov";
+
+  # An optional list containing the order you want profiles to be deployed.
+  # This will take effect whenever you run `deploy` without specifying a profile, causing it to deploy every profile automatically.
+  # Any profiles not in this list will still be deployed (in an arbitrary order) after those which are listed
+  profilesOrder = [ "something" "system" ];
+
+  profiles = {
+    # Definition format shown above
+    system = {};
+    something = {};
+  };
+
+  # ...generic options... (see lower section)
+}
+```
+
+### Deploy
+
+This is the top level attribute containing all of the options for this tool
+
+```nix
+{
+  nodes = {
+    # Definition format shown above
+    my-node = {}; 
+    another-node = {};
+  };
+
+  # ...generic options... (see lower section)
+}
+```
+
+### Generic options
+
+This is a set of options that can be put in any of the above definitions, with the priority being `profile > node > deploy`
+
+```nix
+{
+  # This is the user that deploy-rs will use when connecting.
+  # This will default to your own username if not specified anywhere
+  sshUser = "admin";
+
+  # This is the user that the profile will be deployed to (will use sudo if not the same as above).
+  # If `sshUser` is specified, this will be the default (though it will _not_ default to your own username)
+  user = "root";
+
+  # This is an optional list of arguments that will be passed to SSH.
+  sshOpts = [ "-p" "2121" ];
+
+  # Fast connection to the node. If this is true, copy the whole closure instead of letting the node substitute.
+  # This defaults to `false`
+  fastConnection = false;
+
+  # If the previous profile should be re-activated if activation fails.
+  # this defaults to `true`
+  autoRollback = true;
+
+  # If the node should wait for `deploy` to connect for a second time after activation, to confirm the server has not been ruined.
+  # This defaults to `false`, though it is strongly recommend you activate it if you value safety
```

Comment:
You make it sound like it would be downright stupid to not enable this, but you don't enable it by default. This seems contradictory. It would make more sense to enable it by default, and give a list of reasons you might want to disable it.

Reply:
It's off by default as it's unexpected behaviour, I feel like people might try using this tool to change their SSH port for instance, then get fustrated when it refuses to let them. Maybe if I make it more clear that `deploy-rs` does this then it would be ok to have as a default